spec/controllers/auth_controller_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe AuthController, type: :controller do
+  let(:credentials) do
+    { credentials: { token: 'token123' },
+      extra: { raw_info: { name: 'Arthur', avatra_url: '' } }
+    }
+  end
+
+  before do
+    OmniAuth.config.add_mock(:github, credentials)
+    request.env['omniauth.auth'] = OmniAuth.config.mock_auth[:github]
+  end
+
+  after do
+    OmniAuth.config.mock_auth[:github] = nil
+  end
+
+  describe 'GET #github' do
+    before { get :github }
```

Comment:
I know it might lead to repetition, but I'd prefer to have the method I'm testing called on each example. `before` hooks should just run preconditions. Not a blocker, though.